jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -767,6 +767,232 @@ ecma_builtin_typedarray_prototype_set (ecma_value_t this_arg, /**< this argument
   return ret_val;
 } /* ecma_builtin_typedarray_prototype_set */
 
+/**
+ * TypedArray.prototype's 'toString' single element operation routine based
+ * on the Array.prototype's 'toString' single element operation routine
+ *
+ * See also:
+ *          ECMA-262 v5.1, 15.4.4.2
+ *
+ * @return ecma_value_t value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_op_typedarray_get_to_string_at_index (ecma_object_t *obj_p, /**< this object */
+                                           uint32_t index) /**< array index */
+{
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  ecma_string_t *index_string_p = ecma_new_ecma_string_from_uint32 (index);
+  ecma_value_t index_value = ecma_op_object_get (obj_p, index_string_p);
+
```

Comment:
`index_string_p` can be freed here, so you can replace the two `ecma_deref_ecma_string (index_string_p);` calls by one line.